sdk/ml/azure-ai-ml/azure/ai/ml/_azure_environments.py

```diff
@@ -279,3 +276,17 @@ def _convert_arm_to_cli(arm_cloud_metadata):
             module_logger.warning("Property on cloud not found in arm cloud metadata: %s", ex)
             continue
     return cli_cloud_metadata_dict
+
+def _add_cloud_to_environments(kwargs):
```

Comment:
Method name is to add the cloud but it is returning a boolean, seems odd. Can we make this method raise exception if cloud metadata already exists and add otherwise? Caller can check if cloud metadata exists if not add it or handle the exception appropriately